### PR TITLE
Fixes #20624 - Register dead letter handler from foreman tasks core

### PR DESCRIPTION
**Pull request by adamruzicka**
requires
- [x] https://github.com/theforeman/foreman-tasks/pull/276
- [x] https://github.com/Dynflow/dynflow/pull/245

### CODE REVIEW

**Comment by iNecas** [comment]
Not sure why jenkins is not picking up the task

**Comment by iNecas** [comment]
[test]

**Comment by iNecas** [comment]
Tests are failing, as we renamed the `DeadLetterHandler` to `DeadLetterSilencer`

**Comment by adamruzicka** [comment]
Fixed in https://github.com/theforeman/foreman-tasks/pull/281

File: smart_proxy_dynflow_core.gemspec [hunk 1]
```diff
@@ -30,8 +30,8 @@ Gem::Specification.new do |gem|
   gem.add_development_dependency('rack-test', '~> 0')
   gem.add_development_dependency('rubocop', '0.32.1')
 
-  gem.add_runtime_dependency('dynflow', "~> 0.8.27")
-  gem.add_runtime_dependency('foreman-tasks-core', '>= 0.1.5')
+  gem.add_runtime_dependency('dynflow', "~> 0.8.29")
+  gem.add_runtime_dependency('foreman-tasks-core', '>= 0.1.6')
```
Comment (iNecas) [inline smart_proxy_dynflow_core.gemspec]: Please update to 0.1.7, as the fix for silencer is included there.

File: lib/smart_proxy_dynflow_core/core.rb [hunk 2]
```diff
@@ -73,6 +76,14 @@ def ensure_initialized
         @instance
       end
 
+      def plugin_matchers
```
Comment (iNecas) [inline lib/smart_proxy_dynflow_core/core.rb]: From the name it's not clear, what kind of matchers it is. What about `silencer_matchers`?

File: lib/smart_proxy_dynflow_core/core.rb [hunk 3]
```diff
@@ -73,6 +76,14 @@ def ensure_initialized
...
+        @matchers ||= []
+      end
+
+      def register_plugin_matchers(matchers)
```
Comment (iNecas) [inline lib/smart_proxy_dynflow_core/core.rb]: Similarly to `plugin_matchers`, thinking about improving the naming. What about `register_silencer_matchers`, as we don't really care if it's from plugins, but it should be clean it's about silencing

**Comment by iNecas** [comment]
Tested and works well. Thanks @adamruzicka

**Pull request merged**